### PR TITLE
Update TEST_true and TEST_false

### PR DESCRIPTION
**Pull request by pauli-oracle**
- [x] tests are added or updated

Modify the TEST_true and TEST_false macros so they correctly deal with all integral types.

### CODE REVIEW

**Comment by dot-asm** [comment]
Red crosses from travis and appveyor are relevant and should be addressed.

File: test/testutil.h [hunk 1]
```diff
@@ -291,8 +291,8 @@ void test_info_c90(const char *desc, ...) PRINTF_FORMAT(1, 2);
 # define TEST_mem_eq(a, m, b, n) test_mem_eq(__FILE__, __LINE__, #a, #b, a, m, b, n)
 # define TEST_mem_ne(a, m, b, n) test_mem_ne(__FILE__, __LINE__, #a, #b, a, m, b, n)
 
-# define TEST_true(a)         test_true(__FILE__, __LINE__, #a, a)
-# define TEST_false(a)        test_false(__FILE__, __LINE__, #a, a)
+# define TEST_true(a)         test_true(__FILE__, __LINE__, #a, (a) != 0)
+# define TEST_false(a)        test_false(__FILE__, __LINE__, #a, (a) == 0)
```
Comment (levitte) [inline test/testutil.h]: Unless I gravely misunderstand, both of these test that `(a) != 0`...  that doesn't seem quite right

Comment (pauli-oracle) [inline test/testutil.h]: You're right.  Silly miss on my part.

**Comment by richsalz** [comment]
commit 298f40e in matser.

**Pull request closed without merging**